### PR TITLE
Allow P<->C Chain Transfers

### PR DESCRIPTION
**Pull request by EvanJRichard**
## Summary
Previously, only X<>P and X<>C Chain transfers were legal on mainnet. Now, P<>C transfers are legal - essentially the only difference being the Chain IDs used.  This PR proposes to allow P<>C chain transfers.

This is accomplished by removing some checks that a transfer was always X<>P or X<>C. We could re-add the checks if we want to ensure only X/P/C are used, or we could leave them out if we want imports and exports on (for example) custom test networks to work too.

Additionally, some prompt labels were changed to reflect this update.

### Testing
Added 4 tests - two to P chain (export to C, import from C) and two to C chain(export to P, import from P).
C chain ID for local networks changed. Updated tests where necessary.

### CODE REVIEW

File: src/parser.c [hunk 1]
```diff
@@ -776,10 +776,7 @@ enum parse_rv parse_ImportTransaction(struct ImportTransactionState *const state
       switch (state->state) {
         case 0: // ChainID
             CALL_SUBPARSER(id32State, Id32);
-            if(is_pchain_transaction(meta->type_id)) {
-              if(memcmp(network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
-                REJECT("Invalid XChain ID");
-            } else {
+            if(!is_pchain_transaction(meta->type_id)) {
```
Comment (Ericson2314) [inline src/parser.c]: What does `is_pchain_transaction` mean, in the context of cross-chain transactions?

File: src/network_info.c [hunk 2]
```diff
@@ -25,8 +25,8 @@ const network_info_t network_info[NETWORK_INFO_SIZE] =
   { .network_id = NETWORK_ID_LOCAL,
     // 2eNy1mUFdmaxXNj1eQHUe7Np4gju9sJsEtWQ4MX3ToiNKuADed
     .x_blockchain_id = { 0xd8, 0x91, 0xad, 0x56, 0x05, 0x6d, 0x9c, 0x01, 0xf1, 0x8f, 0x43, 0xf5, 0x8b, 0x5c, 0x78, 0x4a, 0xd0, 0x7a, 0x4a, 0x49, 0xcf, 0x3d, 0x1f, 0x11, 0x62, 0x38, 0x04, 0xb5, 0xcb, 0xa2, 0xc6, 0xbf },
-    // 26sSDdFXoKeShAqVfvugUiUQKhMZtHYDLeBqmBfNfcdjziTrZA
-    .c_blockchain_id = { 0x91, 0x06, 0x0e, 0xab, 0xfb, 0x5a, 0x57, 0x17, 0x20, 0x10, 0x9b, 0x58, 0x96, 0xe5, 0xff, 0x00, 0x01, 0x0a, 0x1c, 0xfe, 0x6b, 0x10, 0x3d, 0x58, 0x5e, 0x6e, 0xbf, 0x27, 0xb9, 0x7a, 0x17, 0x35 },
+    // BZycCYv295rhtyykDkJHaGe7foSadXawXT4XTZPqXBrg
+    .c_blockchain_id = { 0x9d, 0x07, 0x75, 0xf4, 0x50, 0x60, 0x4b, 0xd2, 0xfb, 0xc4, 0x9c, 0xe0, 0xc5, 0xc1, 0xc6, 0xdf, 0xeb, 0x2d, 0xc2, 0xac, 0xb8, 0xc9, 0x2c, 0x26, 0xee, 0xae, 0x6e, 0x6d, 0xf4, 0x50, 0x2b, 0x19 },
```
Comment (Ericson2314) [inline src/network_info.c:29]: Any particular reason this changed?

File: src/parser.c (hunk 1 above)
Comment (Ericson2314) [reply src/parser.c]: That is allows a few different things makes me nervous, I think I will want to inline the case to be more precise.

File: src/parser.c [hunk 3]
```diff
@@ -205,7 +205,7 @@ enum parse_rv parse_SECP256K1TransferOutput(struct SECP256K1TransferOutput_state
                         break;
                     case TRANSACTION_TYPE_ID_PLATFORM_EXPORT:
                         should_break = ADD_PROMPT(
-                            "P to X chain",
+                            "export from P",
```
Comment (Ericson2314) [inline src/parser.c]: Below we do "P chain import" and "C chain export". Should this be in a consistent style?

File: src/network_info.c (hunk 2 above)
Comment (EvanJRichard) [reply src/network_info.c:29]: I can ask around for the "why", if it's useful - I changed it for no reason other than: when I ran a local network using the then-latest `avash` and `avalanchego`, these were the 3 blockchain IDs I got back

File: src/parser.c (hunk 3 above)
Comment (EvanJRichard) [reply src/parser.c]: It should, good catch, fixing

File: src/parser.c (hunk 1 above)
Comment (EvanJRichard) [reply src/parser.c]: This is the code that is called in the "`is-not-pchain-transaction`" branch of the `if` statement that existed before the PR. I'm down to improve the check to be more precise, but what should the case be if not this?

File: src/network_info.c (hunk 2 above)
Comment (EvanJRichard) [reply src/network_info.c:29]: Double-triple checking by asking around internally about the chainID

Comment (EvanJRichard) [reply src/network_info.c:29]: Apparently this should be `2CA6j5zYzasynPsFeNoqWkmTCt3VScMvXUZHbfDJ8k3oGzAPtU`, and not `26sSDdFXoKeShAqVfvugUiUQKhMZtHYDLeBqmBfNfcdjziTrZA` nor `BZycCYv295rhtyykDkJHaGe7foSadXawXT4XTZPqXBrg`. Will fix this and the test cases asap

Comment (Ericson2314) [reply src/network_info.c:29]: OK!

Comment (EvanJRichard) [reply src/network_info.c:29]: Fixed! I got all turned around: `2CA6j5zYzasynPsFeNoqWkmTCt3VScMvXUZHbfDJ8k3oGzAPtU` should be the correct comment here, but the proposed new bytes `0x9d...0x19` are still correct.

File: src/parser.c (hunk 1 above)
Comment (EvanJRichard) [reply src/parser.c]: [cross-annotating off-GitHub conversation to here: John is investigating this and might come back with more questions]

File: src/parser.c [hunk 4]
```diff
@@ -848,10 +845,7 @@ enum parse_rv parse_ExportTransaction(struct ExportTransactionState *const state
     switch (state->state) {
         case 0: // ChainID
             CALL_SUBPARSER(id32State, Id32);
-            if(is_pchain_transaction(meta->type_id)) {
-              if(memcmp(network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
-                REJECT("Invalid XChain ID");
-            } else {
+            if(!is_pchain_transaction(meta->type_id)) {
```
Comment (Ericson2314) [inline src/parser.c]: I think this block should be unconditional. Now that I am more familiar with the code, I think the the old `else` was misleading and what is really going on is that that we were merely doing "else if got by assert" which could just as well apply to the rest of the function.

File: src/parser.c [hunk 5]
```diff
@@ -848,10 +845,7 @@ enum parse_rv parse_ExportTransaction(struct ExportTransactionState *const state
...
               if (is_pchain(state->id32State.buf))
                 meta->swapCounterpartChain = SWAPCOUNTERPARTCHAIN_P;
               else if(!memcmp(network_info_from_network_id_not_null(meta->network_id)->c_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
```
Comment (Ericson2314) [inline src/parser.c]: I will factor out something for this, so it is nicer.

File: src/parser.c [hunk 6]
```diff
@@ -1066,8 +1066,6 @@ enum parse_rv parse_CChainImportTransaction(struct CChainImportTransactionState
       switch (state->state) {
         case 0: // sourceChain
             CALL_SUBPARSER(bidState, blockchain_id_t);
-            if(memcmp(&network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->bidState.buf, sizeof(blockchain_id_t)))
-              REJECT("Invalid XChain ID");
```
Comment (Ericson2314) [inline src/parser.c:1070]: I think we should use `decode_chain_id` to make sure it's still a valid chain here (/ prompt the user to say which one)?

Comment (EvanJRichard) [reply src/parser.c:1070]: We're trying to reduce prompts, so I'll add a `decode` here that just checks that `blockchain_id` is one of the valid blockchain ids

Comment (Ericson2314) [reply src/parser.c:1070]: Sounds good.

File: src/parser.h [hunk 7]
```diff
@@ -401,6 +401,7 @@ enum chain_role {
 
 // aligns with chain_role
 enum SwapCounterpartChain {
+  SWAPCOUNTERPARTCHAIN_X = CHAIN_X,
   SWAPCOUNTERPARTCHAIN_P = CHAIN_P,
   SWAPCOUNTERPARTCHAIN_C = CHAIN_C,
 };
```
Comment (Ericson2314) [inline src/parser.h]: Let's just get rid of this then? it's the same as `enum chain_role` now.

File: src/parser.c [hunk 8]
```diff
@@ -1102,8 +1110,6 @@ enum parse_rv parse_CChainExportTransaction(struct CChainExportTransactionState
       switch (state->state) {
         case 0: // destinationChain
             CALL_SUBPARSER(bidState, blockchain_id_t);
-            if(memcmp(&network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->bidState.buf, sizeof(blockchain_id_t)))
-              REJECT("Invalid XChain ID");
```
Comment (Ericson2314) [inline src/parser.c:1106]: Do the same parse and enure valid here?

**Review by Ericson2314 — APPROVED** [review]

**Pull request merged**